presto-main/src/main/java/io/prestosql/execution/QueryPreparer.java

```diff
@@ -42,17 +44,20 @@
 public class QueryPreparer
 {
     private final SqlParser sqlParser;
+    private final StatementAccessControl statementAccessControl;
 
     @Inject
-    public QueryPreparer(SqlParser sqlParser)
+    public QueryPreparer(SqlParser sqlParser, StatementAccessControl statementAccessControl)
```

Comment:
Pass `StatementAccessControlManager` here. Probably it should also be optional

Reply:
Shall check and update

Reply:
StatementAccessControlManager is being passed.